### ISSUE TITLE
Modification Detection misfiring new dump when no change has happened

### ISSUE THREAD

**Issue by juliamullen** [biothings/biothings.api#85]
The remote file doesn't change but the local file has a new identifier each time. It seems the hub is creating a new local file, realizing it's different from the remote file, and triggering a new dump. It does this, and then creates another new file and repeats the process.

```13:00:00 [26560:MainThread] - dump_biothings_covid19 - DEBUG -- Remote file *** is newer (remote: 1598926471.0, local: 1598842802)
13:00:00 [26560:MainThread] - dump_biothings_covid19 - INFO -- 1 file(s) to download
13:00:02 [29405:MainThread] - dump_biothings_covid19 - DEBUG -- Downloading *** as *** 
13:00:03 [26560:MainThread] - dump_biothings_covid19 - INFO -- biothings_covid19 successfully downloaded
13:00:03 [26560:MainThread] - dump_biothings_covid19 - INFO -- success [steps=dump,post]
14:00:02 [26560:MainThread] - dump_biothings_covid19 - DEBUG -- Remote file *** is newer (remote: 1598926471.0, local: 1598904003)
14:00:02 [26560:MainThread] - dump_biothings_covid19 - INFO -- 1 file(s) to download
14:00:03 [29406:MainThread] - dump_biothings_covid19 - DEBUG -- Downloading *** as *** 
14:00:05 [26560:MainThread] - dump_biothings_covid19 - INFO -- biothings_covid19 successfully downloaded
14:00:05 [26560:MainThread] - dump_biothings_covid19 - INFO -- success [steps=dump,post]
15:00:00 [26560:MainThread] - dump_biothings_covid19 - DEBUG -- Remote file *** is newer (remote: 1598926471.0, local: 1598907605)
15:00:00 [26560:MainThread] - dump_biothings_covid19 - INFO -- 1 file(s) to download
15:00:01 [29403:MainThread] - dump_biothings_covid19 - DEBUG -- Downloading *** as *** 
15:00:03 [26560:MainThread] - dump_biothings_covid19 - INFO -- biothings_covid19 successfully downloaded
15:00:03 [26560:MainThread] - dump_biothings_covid19 - INFO -- success [steps=dump,post]
16:00:00 [26560:MainThread] - dump_biothings_covid19 - DEBUG -- Remote file *** is newer (remote: 1598926471.0, local: 1598911202)
16:00:00 [26560:MainThread] - dump_biothings_covid19 - INFO -- 1 file(s) to download
16:00:01 [29406:MainThread] - dump_biothings_covid19 - DEBUG -- Downloading *** as *** 
16:00:03 [26560:MainThread] - dump_biothings_covid19 - INFO -- biothings_covid19 successfully downloaded
16:00:03 [26560:MainThread] - dump_biothings_covid19 - INFO -- success [steps=dump,post]
17:00:01 [26560:MainThread] - dump_biothings_covid19 - DEBUG -- Remote file *** is newer (remote: 1598926471.0, local: 1598914803)
17:00:01 [26560:MainThread] - dump_biothings_covid19 - INFO -- 1 file(s) to download
17:00:03 [29403:MainThread] - dump_biothings_covid19 - DEBUG -- Downloading *** as ***

**Comment by zcqian** [comment]
From the logs it seems that the remote file, while it does not change, is always newer than the local time, hence the behavior. We can elect to use the remote modification time instead of relying on the local time when creating downloaded files, if this situation becomes an issue again. I will leave this issue open but change the tag to enhancement, if that's okay with you.
@juliamullen @newgene

**Comment by newgene** [comment]
@zcqian if both timestamps have tz info included, we should be able to compare them correctly?

**Comment by zcqian** [comment]
@newgene Technically file timestamps are always stored in UTC and `Last-Modifled` is always in GMT. I guess our code treats `Last-Modified` as local time (which makes it seven or eight hours ahead of its actual time), but I'm not exactly sure. I will take a look and check.

**Comment by zcqian** [comment]
Closed via PR #167
